GVFS/GVFS.UnitTests.Windows/Windows/Upgrader/NewProductUpgradeTests.cs

```diff
@@ -0,0 +1,102 @@
+
+            productUpgrader.ShouldNotBeNull();
+            productUpgrader.ShouldBeOfType<NuGetUpgrader>();
+        }
+
+        [TestCase]
+        public void CreatesGitHubWhenConfigured()
+        {
+            this.gvfsConfig = this.BuildGitHubtGvfsConfig();
+            ProductUpgrader.TryCreateUpgrader(this.tracer, this.fileSystemMock.Object, this.gvfsConfig, this.credentialStoreMock.Object, false, false, out ProductUpgrader productUpgrader, out string error);
+
+            productUpgrader.ShouldNotBeNull();
+            productUpgrader.ShouldBeOfType<GitHubUpgrader>();
+        }
+
+        [TestCase]
+        public void NoUpgraderWhenNuGetFeedMissing()
+        {
+            this.gvfsConfig = this.BuildNuGetConfigMissingUpgradeFeed();
+            ProductUpgrader.TryCreateUpgrader(this.tracer, this.fileSystemMock.Object, this.gvfsConfig, this.credentialStoreMock.Object, false, false, out ProductUpgrader productUpgrader, out string error);
+
+            productUpgrader.ShouldBeNull();
+        }
+
+        [TestCase]
+        public void NoUpgraderWhenNuGetPackNameMissing()
+        {
+            this.gvfsConfig = this.BuildNuGetConfigMissingPackageName();
+            ProductUpgrader.TryCreateUpgrader(this.tracer, this.fileSystemMock.Object, this.gvfsConfig, this.credentialStoreMock.Object, false, false, out ProductUpgrader productUpgrader, out string error);
+
+            productUpgrader.ShouldBeNull();
+        }
+
+        private MockLocalGVFSConfig BuildNuGetGvfsConfig()
+        {
+            MockLocalGVFSConfig gvfsConfig = new MockLocalGVFSConfig();
+            gvfsConfig.TrySetConfig(GVFSConstants.LocalGVFSConfig.UpgradeRing, "slow", out _);
+            gvfsConfig.TrySetConfig(GVFSConstants.LocalGVFSConfig.UpgradeFeedPackageName, "package", out _);
+            gvfsConfig.TrySetConfig(GVFSConstants.LocalGVFSConfig.UpgradeFeedUrl, "https://pkgs.dev.azure.com/contoso/", out _);
+            return gvfsConfig;
+        }
+
+        private MockLocalGVFSConfig BuildNuGetConfigMissingUpgradeFeed()
+        {
+            MockLocalGVFSConfig gvfsConfig = new MockLocalGVFSConfig();
+            gvfsConfig.TrySetConfig(GVFSConstants.LocalGVFSConfig.UpgradeRing, "slow", out _);
+            gvfsConfig.TrySetConfig(GVFSConstants.LocalGVFSConfig.UpgradeFeedPackageName, "package", out _);
+            return gvfsConfig;
+        }
+
+        private MockLocalGVFSConfig BuildNuGetConfigMissingPackageName()
+        {
+            MockLocalGVFSConfig gvfsConfig = new MockLocalGVFSConfig();
+            gvfsConfig.TrySetConfig(GVFSConstants.LocalGVFSConfig.UpgradeRing, "slow", out _);
+            gvfsConfig.TrySetConfig(GVFSConstants.LocalGVFSConfig.UpgradeFeedUrl, "https://pkgs.dev.azure.com/contoso/", out _);
+            return gvfsConfig;
+        }
+
+        private MockLocalGVFSConfig BuildGitHubtGvfsConfig()

```

Comment:
Typo: `Hubt`.   The other methods don't include "Gvfs" in the name, so you could consider renaming this method to `BuildGitHubConfig`.